doc/source/reference/arrays.classes.rst

```diff
@@ -73,40 +72,26 @@ NumPy provides several hooks that classes can customize:
    If an :class:`ndarray` subclass defines the :func:`__array_ufunc__`
    method, this disables the :func:`__array_wrap__`,
    :func:`__array_prepare__`, :data:`__array_priority__` mechanism
-   described below.
+   described below (which may eventually be deprecated).  Furthermore, it
+   takes precedence over right-hand binary operator override methods (such
+   as ``__rmul__``) or comparison operations (such as ``__gt__``) in your
+   class: the presumption is that your class will already be called first
+   via :func:`__array_ufunc__`.
+
+   Like with other methods in python, such as ``__hash__`` and
+   ``__iter__``, it is possible to indicate that your class does *not*
+   support ufuncs by setting ``__array_ufunc__ = None`` (assuming your
+   class is not an :class:`ndarray` subclass). With this, inside ufuncs,
+   your class will be treated as if it returned :obj:`NotImplemented`
```

Comment:
I don't think this is correct. It is the calling class, ndarray and subclasses thereof, for which it is equivalent to calling `NotImplemented`. The effect is that unary ops fail and binary ops get passed to the corresponding right op. Or am I missing something?